monitor/services/src/main/java/org/datacleaner/monitor/server/controllers/ComponentControllerV1.java

```diff
@@ -210,18 +207,41 @@ public OutputColumns getOutputColumns(@PathVariable(PARAMETER_NAME_TENANT) final
     @RequestMapping(value = "/{name}", method = RequestMethod.PUT, consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
     public ProcessStatelessOutput processStateless(@PathVariable(PARAMETER_NAME_TENANT) final String tenant,
             @PathVariable(PARAMETER_NAME_NAME) final String name,
+            @RequestParam(value = PARAMETER_NAME_OUTPUT_FORMAT, required = false, defaultValue = "") String outputFormat,
             @RequestBody final ProcessStatelessInput processStatelessInput) {
         String decodedName = ComponentsRestClientUtils.unescapeComponentName(name);
         logger.debug("One-shot processing '{}'", decodedName);
         TenantContext tenantContext = _tenantContextFactory.getContext(tenant);
         ComponentHandler handler = componentHandlerFactory.createComponent(tenantContext, decodedName, processStatelessInput.configuration);
         ProcessStatelessOutput output = new ProcessStatelessOutput();
-        output.rows = getJsonNode(handler.runComponent(processStatelessInput.data, _maxBatchSize));
+        output.rows = getOutputJsonNode(handler, handler.runComponent(processStatelessInput.data, _maxBatchSize), outputFormat);
         output.result = getJsonNode(handler.closeComponent());
 
         return output;
     }
 
+    private JsonNode getOutputJsonNode(ComponentHandler handler, Collection<List<Object[]>> data, String outputFormat) {
+        if("columnMap".equals(outputFormat)) {
+            org.datacleaner.api.OutputColumns columns = handler.getOutputColumns();
+            int columnCount = columns.getColumnCount();
+            List<List<Map<String, Object>>> columnMapOutput = new ArrayList<>(data.size());
+            for(List<Object[]> rowGroup: data) {
+                List<Map<String, Object>> columnMapRowGroup = new ArrayList<>(rowGroup.size());
+                for(Object[] row: rowGroup) {
+                    Map<String, Object> columMapRow = new HashMap<>(columnCount);
+                    for(int i = 0; i < columnCount; i++) {
+                        columMapRow.put(columns.getColumnName(i), row[i]);
```

Comment:
I could imagine cases where legal DC column names could end up as illegal JSON, though I must admit I'm not sure.